3.10/aql/operations-replace.md

```diff
@@ -1,18 +1,26 @@
 ---
 layout: default
-description: The REPLACE keyword can be used to completely replace documents in a collection
+description: >-
+  You can use `REPLACE` operations to completely replace documents with all
+  their attributes in a collection
```

Comment:
```suggestion
  their non-system attributes in a collection
```

Reply:
This is getting out of hand... The part about system attributes was merely a side note, and it's mentioned five times on the page now. I don't see a need to mention it in the lead paragraph, too, which is only meant as a synopsis that can be inaccurate for the sake of providing a general overview. It's also incorrect to say that non-system attributes are excluded (_from and _to are system attributes but can be modified for a long time).

Reply:
I see no harm in repeating the same information to avoid that the user misunderstands it. This is not a poem, it doesn't need to be nice, it has to be unambiguous. Not every user reads the whole page every time she needs to recall something, I guess most people never read the whole page. I want to avoid that a user reads the synopsis ("replace documents with all their attributes"), tries to replace all the attributes (how many people would check further?), this doesn't work, the user searches for a bug in her code for hours, asks the support - just to learn that the sentence in the synopsis is inaccurate.

My suggestion does not say that system (you mean system, not non-system) attributes are excluded, it says that non-system attributes are replaced. It doesn't say anything about system attributes at all (a simplification for the synopsis but nothing incorrect).

Reply:
I meant to write:
> It's incorrect to say that system attributes are excluded

"It replaces documents with all their non-system attributes" or "It replaces documents with all their attributes excluding system attributes" is the same statement. Depending on the definition of system attributes,
- every attribute that starts with an underscore is a system attribute
- only top-level attributes that start with an underscore are system attributes
- only specific attributes are system attributes, namely `_key`, `_id`, `_rev`, `_from`, and `_to`, for which there is a special treatment in code and the API
- only specific attributes are system attributes, namely `_key`, `_id`, and `_rev`, which are immutable (`_key`), virtual attributes (`_id`), or not user-controlled (`_rev`).

This is not properly defined, and you can find inconsistent behavior concerning system attributes in ArangoDB. If you don't expect users to read more than the subheading, then we need to clearly state what it does and doesn't replace in the synopsis. It does replace _from, it does replace _to (if provided by the user), it does not replace the other three, but that is not directly related to the AQL UPDATE and REPLACE operations.